tests/director/tests/application/application_templates_api_test.go

```diff
@@ -1493,6 +1493,70 @@ func TestRegisterApplicationFromTemplate(t *testing.T) {
 	example.SaveExample(t, createAppFromTmplRequest.Query(), "register application from template")
 }
 
+func TestRegisterApplicationFromTemplateWithSystemFieldDiscoveryLabel(t *testing.T) {
+	//GIVEN
+	ctx := context.TODO()
+	nameJSONPath := "$.name-json-path"
+	displayNameJSONPath := "$.display-name-json-path"
+	appTemplateName := fixtures.CreateAppTemplateName("template")
+	appTmplInput := fixtures.FixAppTemplateInputWithDefaultDistinguishLabel(appTemplateName, conf.SubscriptionConfig.SelfRegDistinguishLabelKey, conf.SubscriptionConfig.SelfRegDistinguishLabelValue)
+	appTmplInput.ApplicationInput.Description = ptr.String("test {{display-name}}")
+	regionInConfiguration := "eu-1"
+	appTmplInput.ApplicationInput.Labels["region"] = regionInConfiguration
+	appTmplInput.Placeholders = []*graphql.PlaceholderDefinitionInput{
+		{
+			Name:        "name",
+			Description: ptr.String("name"),
+			JSONPath:    &nameJSONPath,
+		},
+		{
+			Name:        "display-name",
+			Description: ptr.String("display-name"),
+			JSONPath:    &displayNameJSONPath,
+		},
+	}
+	tenantId := tenant.TestTenants.GetDefaultSubaccountTenantID()
+
+	appTmplInput.Labels["global_subbacount_id"] = tenantId
+	appTmplInput.Labels["systemFieldDiscovery"] = true
+
+	appTmpl, err := fixtures.CreateApplicationTemplateFromInput(t, ctx, certSecuredGraphQLClient, tenantId, appTmplInput)
+	defer fixtures.CleanupApplicationTemplate(t, ctx, certSecuredGraphQLClient, tenantId, appTmpl)
+	require.NoError(t, err)
+	require.Equal(t, conf.SubscriptionConfig.SelfRegRegion, appTmpl.Labels[tenantfetcher.RegionKey])
+	require.Equal(t, tenantId, appTmpl.Labels["global_subbacount_id"])
+	require.Equal(t, true, appTmpl.Labels["systemFieldDiscovery"])
+
+	appFromTmpl := graphql.ApplicationFromTemplateInput{TemplateName: appTemplateName, Values: []*graphql.TemplateValueInput{
+		{
+			Placeholder: "name",
+			Value:       "new-name",
+		},
+		{
+			Placeholder: "display-name",
+			Value:       "new-display-name",
+		}}}
+	appFromTmplGQL, err := testctx.Tc.Graphqlizer.ApplicationFromTemplateInputToGQL(appFromTmpl)
+	require.NoError(t, err)
+	createAppFromTmplRequest := fixtures.FixRegisterApplicationFromTemplate(appFromTmplGQL)
+	outputApp := graphql.ApplicationExt{}
+	//WHEN
+	err = testctx.Tc.RunOperationWithCustomTenant(ctx, certSecuredGraphQLClient, tenantId, createAppFromTmplRequest, &outputApp)
+	defer fixtures.UnregisterApplication(t, ctx, certSecuredGraphQLClient, tenantId, outputApp.ID)
+
+	//THEN
+	require.NoError(t, err)
+	require.NotEmpty(t, outputApp)
+
+	require.NotEmpty(t, outputApp.Webhooks)
+	// webhooks with types - configuration changed and system field discovery
+	require.Len(t, outputApp.Webhooks, 2)
```

Comment:
We can also check the types of the webhooks so we are 100% sure that these 2 webhooks are of type with want.
Also, we can modify the other test with the same assertion logic (one assertion function used on both places would look great)